### PR TITLE
[Docs] Improve Handedness by EEPROM documentation

### PR DESCRIPTION
**Pull request by filterpaper**
## Description

* Add title for all 3 bootloader targets
* Document QMK toolbox support

## Types of Changes

- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: docs/feature_split_keyboard.md [hunk 1]
```diff
@@ -130,24 +130,29 @@ To enable this method, add the following to your `config.h` file:
...
-* `:dfu-util-split-right`
+Next, you will have to flash the EEPROM files once for the correct hand to the controller on each halve. You can do this manually with the following bootloader targets while flashing the firmware:
+
+* Pro micro stock controllers with Caterina bootloader:
+  * `:avrdude-split-left`
+  * `:avrdude-split-right`
+* Pro micro compatible controllers (like Elite-C) with DFU bootloader:
+  * `:dfu-split-left`
+  * `:dfu-split-right`
+* ARM controllers (like Proton-C) with DFU bootloader:
```
Comment (drashna) [inline docs/feature_split_keyboard.md:141]: Should probably note that you need to reset the eeprom after doing this, for dfu-util boards.

Comment (filterpaper) [reply docs/feature_split_keyboard.md:141]: I'm not familiar with split hardware using dfu-util board. Can you clarify the process for me to put that down in the description?

1. Flash with `:dfu-util-split-left`
2. Use QMK Toolbox to `Reset EEPROM` ?

Comment (drashna) [reply docs/feature_split_keyboard.md:141]: the dfu-util split command works by adding a define to specify left vs right, but this only gets set when you init the eeprom.  

Unfortunately, the toolbox can't be used for this.  You'd have to use the `EEP_RST` keycode, use bootmagic lite when flashing, or increment the eeconfig magic number. 

You can see this, here: 
https://github.com/qmk/qmk_firmware/blob/3dee6e28dd2fce27945bd464f5a80d3508e7c1d4/quantum/eeconfig.c#L60-L68

(eg, this only gets called on eeprom init/clear)

Comment (filterpaper) [reply docs/feature_split_keyboard.md:141]: Section updated with a note for `dfu-util`.

**Review by drashna — APPROVED** [review]

File: docs/feature_split_keyboard.md [hunk 2]
```diff
@@ -130,24 +130,31 @@ To enable this method, add the following to your `config.h` file:
...
 
-However, you'll have to flash the EEPROM files for the correct hand to each controller.  You can do this manually, or there are targets for avrdude and dfu to do this, while flashing the firmware: 
-
-* `:avrdude-split-left`
-* `:avrdude-split-right`
-* `:dfu-split-left`
-* `:dfu-split-right`
-* `:dfu-util-split-left`
-* `:dfu-util-split-right`
+Next, you will have to flash the EEPROM files once for the correct hand to the controller on each halve. You can do this manually with the following bootloader targets while flashing the firmware:
```
Comment (daskygit) [inline docs/feature_split_keyboard.md:133]: For controllers using emulated eeprom these need to be specified every time you flash.

**Comment by daskygit** [comment]
Would it also be worth either switching the example to use `qmk flash` or adding it as an example?

File: docs/feature_split_keyboard.md (hunk 2 above)
Comment (filterpaper) [reply docs/feature_split_keyboard.md:133]: > For controllers using emulated eeprom these need to be specified every time you flash.

I presume this apply to ARM controllers?

**Comment by filterpaper** [comment]
> Would it also be worth either switching the example to use `qmk flash` or adding it as an example?

@drashna I do find the `qmk` command clearer. However `make` is recommended as the template for all keyboard readme file and the original eeprom handedness command example is consistent with that. What do you think?

File: docs/feature_split_keyboard.md (hunk 2 above)
Comment (daskygit) [reply docs/feature_split_keyboard.md:133]: Yeah a few of them anyway.

Comment (filterpaper) [reply docs/feature_split_keyboard.md:133]: Added a note to cover this fact in the ARM controller section that follows.

File: docs/feature_split_keyboard.md [hunk 3]
```diff
@@ -130,24 +130,31 @@ To enable this method, add the following to your `config.h` file:
...
+
+* Pro micro stock controllers with Caterina bootloader:
```
Comment (daskygit) [inline docs/feature_split_keyboard.md:135]:
```suggestion
* AVR controllers with the Caterina bootloader (e.g. Pro Micro):
```

File: docs/feature_split_keyboard.md [hunk 4]
```diff
@@ -130,24 +130,31 @@ To enable this method, add the following to your `config.h` file:
...
+  * `:avrdude-split-left`
+  * `:avrdude-split-right`
+* Pro micro compatible controllers (like Elite-C) with DFU bootloader:
+  * `:dfu-split-left`
+  * `:dfu-split-right`
+* ARM controllers (like Proton-C) with DFU bootloader:
```
Comment (daskygit) [inline docs/feature_split_keyboard.md:141]:
```suggestion
* ARM controllers with a DFU compatible bootloader (e.g. Proton-C):
```

File: docs/feature_split_keyboard.md [hunk 5]
```diff
@@ -130,24 +130,31 @@ To enable this method, add the following to your `config.h` file:
...
-* `:dfu-split-left`
-* `:dfu-split-right`
-* `:dfu-util-split-left`
-* `:dfu-util-split-right`
+Next, you will have to flash the EEPROM files once for the correct hand to the controller on each halve. You can do this manually with the following bootloader targets while flashing the firmware:
+
+* Pro micro stock controllers with Caterina bootloader:
+  * `:avrdude-split-left`
+  * `:avrdude-split-right`
+* Pro micro compatible controllers (like Elite-C) with DFU bootloader:
```
Comment (daskygit) [inline docs/feature_split_keyboard.md:138]:
```suggestion
* AVR controllers with the stock Amtel DFU or DFU compatible bootloader (e.g. Elite-C):
```

**Review by daskygit — CHANGES_REQUESTED** [review]

**Review by daskygit — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Comment by drashna** [comment]
Thanks!

**Pull request merged**